### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by FDelor**
On branch master
 Initial commit

### CODE REVIEW

File: CodeReviewTest/Faktory.cpp [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#include "Faktorie.h"
+
+Faktorie::Faktorie()
```
Comment (CSchmiedl) [inline CodeReviewTest/Faktory.cpp]: very factorial

File: CodeReviewTest/Faktory.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
+#ifndef FAKTORIY_H
+#define FAKTORIY_H
+
+#include <QSharedPointer>
+#include <QWeakPointer>
+#include <QHash>
```
Comment (CSchmiedl) [inline CodeReviewTest/Faktory.h:6]: do you really need that hash?

Comment (CSchmiedl) [reply CodeReviewTest/Faktory.h:6]: solved

File: CodeReviewTest/Faktory.h [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+template<typename Type,typename KeyType>
+class Faktory
+{
+public:
+
+    Faktory(){}
+    ~Factory(){
+        m_instances.clear();
+    }
+
```
Comment (CSchmiedl) [inline CodeReviewTest/Faktory.h:17]: blub

**Review by CSchmiedl — CHANGES_REQUESTED** [review]

**Review by FDelor — COMMENTED** [review]
we realy need this

**Comment by FDelor** [comment]
i will create new one

**Comment by FDelor** [comment]
nee

**Review by FDelor — COMMENTED** [review]
new changes

**Comment by FDelor** [comment]
ffffsdfgs

**Comment by CSchmiedl** [comment]
asdf

**Review by CSchmiedl — APPROVED** [review]

**Pull request merged**